src/utils/focusTrap.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * @copyright Copyright (c) 2022 John Molakvoæ <skjnldsv@protonmail.com>
+ *
+ * @author John Molakvoæ <skjnldsv@protonmail.com>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+// eslint-disable-next-line import/named
+import { FocusTrap } from 'focus-trap'
+
+/**
+ * Return the default global focus trap stack
+ *
+ * @return {FocusTrap[]}
+ */
+export const getTrapStack = function() {
+	// Create global stack if undefined
+	Object.assign(window, { _nc_focus_trap: window._nc_focus_trap || [] })
```

Comment:
Does this have to be done by every app ?
Wouldn't it make sense to put the global focus trap logic in `@nextcloud/focus-trap` ? This would ensure that no app forgets to use the global focus trap.

Reply:
Looking at `NcPopover` and `NcModal` we already have it twice in this repo.

Reply:
> Wouldn't it make sense to put the global focus trap logic in `@nextcloud/focus-trap` ? This would ensure that no app forgets to use the global focus trap.

Is there any app manually using a focus trap?
If not I guess we're good with this in that library then.

Because we already have a hard time maintaining all those libraries. I don't want to add one more to my list :(

Reply:
Also, having an external lib to build and release and maintain deps just to have that wrapper is quite overkill :(

Reply:
I would delete @nextcloud/focus-trap once we have this in and working tbh

Reply:
Make sense :+1: